core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -302,6 +302,7 @@ def preprocess(self, X, preprocess_nonadaptive=True, preprocess_stateful=True, *
         Preprocesses the input data into internal form ready for fitting or inference.
         It is not recommended to override this method, as it is closely tied to multi-layer stacking logic. Instead, override `_preprocess`.
         """
+        X = copy.deepcopy(X)
```

Comment:
Please rebase with mainline and then remove this. I fixed the bug in https://github.com/awslabs/autogluon/pull/2309

Thanks a ton for figuring out that there was an issue!